harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -9,8 +10,12 @@
 
 
 @pytest.fixture(scope="session")
-def focal_image_url():
-    return "http://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img"  # noqa
+def focal_image_url(request):
+    base_url = request.config.getoption(
+        '--image-cache-url',
+        'http://cloud-images.ubuntu.com/releases/focal/release'
+        )
```

Comment:
I just look over again, there will have a potential bug, the default value of `--image-cache-url` is `''`, then `base_url = ''`
And the design of `urljoin`:
```python
>>> urljoin('http://abc/a', '123')
'http://abc/123'
>>> urljoin('http://abc/a', '/123')
'http://abc/123'
>>> urljoin('http://abc/a', '123')
'http://abc/123'
>>> urljoin('https://abc/a/', '123')
'https://abc/a/123'
>>> urljoin('https://abc/a//', '/123')
'https://abc/123'
>>> urljoin('https://abc/a/', '/123')
'https://abc/123'

```

ref: https://github.com/harvester/tests/blob/main/harvester_e2e_tests/fixtures/images.py#L18

Reply:
Nice comment, patch submitted referring to `images.py` and tested on harvester-runtests#326.

A quick manual check:
```
>>> def focal_image_url(base_url):
...    base_url = base_url.strip()
...    base_url = base_url or "http://cloud-images.ubuntu.com/releases/focal/release"
...    return urljoin(f"{base_url}/", "ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img")
... 
>>> focal_image_url("http://abc")
'http://abc/ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img'
>>> focal_image_url("http://abc") == focal_image_url("http://abc/")
True

>>> focal_image_url("http://abc/a")
'http://abc/a/ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img'
>>> focal_image_url("http://abc/a") == focal_image_url("http://abc/a/")
True

>>> focal_image_url("")
'http://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img'
>>> focal_image_url("") == focal_image_url(" ")
True
```